snapcraft/internal/lxd/_project.py

```diff
@@ -158,13 +159,14 @@ def refresh(self):
             self._container_run(['apt-get', 'upgrade', '-y'])
             self._container_run(['snap', 'refresh'])
 
-    def clean(self, parts, step):
+    def clean(self, parts: List[str], step: str):
         # clean with no parts deletes the container
         if not step:
-            if self._get_container_status():
-                print('Deleting {}'.format(self._container_name))
-                subprocess.check_call([
-                    'lxc', 'delete', '-f', self._container_name])
+            if not parts:
+                if self._get_container_status():
+                    print('Deleting {}'.format(self._container_name))
```

Comment:
I know this wasn't introduced here, but should this be using a logger? Or `click.echo` at the very least?

Reply:
There is a reason for this, and it is stated in the original PR, I just cannot recall it now.

That said, I wouldn't block further review of this PR based on this

Reply:
Sounds like a good opportunity for a comment there, then :wink: . I am currently testing.